File: euler/clojure/Euler2.clj
(defn solve-it [max]
  (loop [ths 1, nxt 1, total 0]
    (if (>= ths max)
      (println total)
      (if (zero? (rem ths 2))
        (recur nxt (+ nxt ths) (+ total ths))
        (recur nxt (+ nxt ths) total)
      )
    )
  )
)

(solve-it 4000000)

File: euler/ruby/Euler2.rb
def fibonacciTo(max)
  thisNum, nextNum = 1, 1
  while thisNum < max
    yield thisNum
    thisNum, nextNum = nextNum, nextNum + thisNum
  end
end

total = 0
fibonacciTo(4000000) { |n| total += n if n.remainder(2) == 0 }
print total